midend/flattenInterfaceStructs.h

```diff
@@ -55,7 +61,7 @@ struct StructTypeReplacement : public IHasDbPrint {
     // Maps internal fields names to types.
     // .t -> T
     // .t.s -> S
-    std::map<cstring, const IR::Type_Struct*> structFieldMap;
+    std::map<cstring, const IR::Type*> structFieldMap;
```

Comment:
These can probably be Type_StructLike

Reply:
> @hesingh Do you have an ETA for this PR? I can chip in with implementation if you need a bit more help. I would need this feature is near future as well.

@hanw: Today we could complete the code to support P4 header with nested-bit vector struct and references.  issue363-bmv2.p4 passed testing with all code being exercised.  Next, we have to deal with 88 failed p4c tests - should be a day or two.  Then, we have to develop more tests to test the language changes due to this PR due to Mihai's comments.  The code changes have taken care of rest of Mihai's comments.  Will try to complete work so that reviewers can approve by Friday of this week.

Reply:
> These can probably be Type_StructLike

We have change the flatten pass so that the .h file API are homogenous - all API changed to use IR:Type.  Then code has been changed in the .cpp to use StructLike.  I will unicast diffs in email for quick review because we won't update the PR until 88 failed tests pass.